app/src/main/java/com/example/notes/model/NoteRepository.kt

```diff
@@ -0,0 +1,8 @@
+package com.example.notes.model
+
+interface NoteRepository {
+        fun insertNote(noteData: NoteData, callback: (status: Boolean) -> Unit) : Boolean
+        fun readAllNote(callback: (notes: List<NoteData>) -> Unit) : Boolean
+        fun removeNote(noteID: Int, callback: (status: Boolean) -> Unit) : Boolean
+        fun updateNote(noteData: NoteData, callback: (status: Boolean) -> Unit) : Boolean
+}
```

Comment:
@AdiTelem I would use sealed class to encapsulate response, for example:
```
sealed class NetworkResult<T>(
    val data: T? = null,
    val message: String? = null
) {

    class Success<T>(data: T) : NetworkResult<T>(data)

    class Error<T>(message: String?, data: T? = null) : NetworkResult<T>(data, message)

}
```

So you can create your  response that will holds already the data you want to return
Like  : 

`NetworkResult.Success<Boolean>(true)`
`NetworkResult.Failure< NoteData >(message = errorMessage)`

Reply:
note taken, will be back logged and addressed  in further development